spec/integration/email_outbound_spec.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+# We already have Email::Sender and Email::MessageBuilder specs along
+# with mailer specific mailer specs like UserEmail, but sometimes we need
+# to test things along the whole outbound flow including the MessageBuilder
+# and the Sender.
+describe "Outbound Email" do
+  def send_email(opts = {})
+    message = TestMailer.send_test("test@test.com", opts)
+    result = Email::Sender.new(message, :test_message).send
+    [message, result]
+  end
+
+  context "email custom headers" do
+    it "discards the custom header if it is one that we have already set based on arguments" do
+      SiteSetting.email_custom_headers = "Precedence: bulk"
+      post = Fabricate(:post)
+      message, result = send_email(post_id: post.id, topic_id: post.topic_id)
+      expect(message.header["Precedence"].value).to eq("list")
+    end
+
+    it "does not discard unique custom headers" do
```

Comment:
I'm not sure if this description is accurate. If this case we're not really discarding the unique custom headers but rather that the custom headers are being set.

Reply:
Yeah makes sense will update